vision/src/autogluon/vision/image_classification/image_classification.py

```diff
@@ -1,341 +1,256 @@
+        random_state : numpy.random.state
+            The random_state for shuffling, only used if `val_data==None`.
+            Note that the `random_state` only affect the splitting process, not model training.
+        time_limit : int
+            Time limit in seconds, default is 12 hours. If `time_limit` is hit during `fit`, the
+            HPO process will interupt and return the current best configuration.
+        epochs : int
+            The `epochs` for model training, if `None` is provided, then default `epochs` for model
+            will be used.
+        num_trials : int, default is 1
+            The number of HPO trials. If `None`, will run only one trial.
+        hyperparameters : dict
+            Extra hyperparameters for specific models.
+        search_strategy : str
+            Searcher strategy for HPO, 'random' by default.
+        scheduler_options : dict
+            Extra options for HPO scheduler, please refer to `autogluon.Searcher` for details.
+        nthreads_per_trial : int
+            Number of CPU threads for each trial, if `None`, will detect the # cores on current instance.
+        ngpus_per_trial : int
+            Number of GPUs to use for each trial, if `None`, will detect the # gpus on current instance.
+        dist_ip_addrs : list
+            If not `None`, will spawn tasks on distributed nodes.
+
+        """
+        use_rec = False
+        if isinstance(train_data, str) and train_data == 'imagenet':
+            logging.warn('ImageNet is a huge dataset which cannot be downloaded directly, ' +
+                         'please follow the data preparation tutorial in GluonCV.' +
+                         'The following record files(symlinks) will be used: \n' +
+                         'rec_train : ~/.mxnet/datasets/imagenet/rec/train.rec\n' +
+                         'rec_train_idx : ~/.mxnet/datasets/imagenet/rec/train.idx\n' +
+                         'rec_val : ~/.mxnet/datasets/imagenet/rec/val.rec\n' +
+                         'rec_val_idx : ~/.mxnet/datasets/imagenet/rec/val.idx\n')
+            train_data = pd.DataFrame({'image': [], 'label': []})
+            val_data = pd.DataFrame({'image': [], 'label': []})
+            use_rec = True
+        if isinstance(train_data, str):
+            from d8.image_classification import Dataset as D8D
+            names = D8D.list()
+            if train_data.lower() in names:
+                train_data = D8D.get(train_data)
+            if val_data is None:
+                train_data, val_data = train_data.split(train_size)
+        if isinstance(val_data, str):
+            from d8.image_classification import Dataset as D8D
+            names = D8D.list()
+            if val_data.lower() in names:
+                val_data = D8D.get(val_data)
+        if self._classifier is not None:
+            self._fit_summary = self._classifier.fit(train_data, val_data, train_size, random_state, resume=False)
+            return
```

Comment:
return self will be compatible with previous `xxx = task.fit()` usage, though not necessary any more

Reply:
`xxx = task.fit()` is confusing now though, because it makes people think that `xxx` and `task` are different things (thats what I thought the first time I read through the website documentation).